packages/plugin/src/classes/helpers/classes.js

```diff
@@ -69,13 +69,50 @@ export function convertClassToUI5Extend(
     const memberName = member.key.name;
 
     if (t.isClassMethod(member)) {
+      const isConstructor = member.kind === "constructor";
+      const membersToAssign = [];
+      const params = isConstructor
+        ? member.params?.map((param) => {
+            // handling of parameter properties for constructors (TypeScript):
+            // https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties
+            //   -> extracting the real parameters and store the members to assign
+            if (param.type === "TSParameterProperty") {
+              membersToAssign.push(param.parameter);
```

Comment:
Does this discard all information about the parameter property except the parameter itself? As far as I see, a TSParameterProperty also has: accessibility ("public" | "private" | "protected"), decorators, override, readonly

I'd still love to see this merged as-is, but (as witnessed by the underlying issue report) in the long run the expectation will be that every language construct of TypeScript related to class definition will be properly transformed. We'll have to check this in a systematic way - or be faster with allowing ES6 classes in UI5 apps.

Reply:
The other PR #100 changes the way the plugin is executed. It then runs in the Class visitor exit phase and therefore, the TypeScript plugin might have been run already. IMO, it is not the plugin which should take care about this, furthermore it should be the TypeScript plugin upfront.